packages/rocketchat-api/server/v1/settings.js

```diff
@@ -73,10 +73,24 @@ RocketChat.API.v1.addRoute('settings/:_id', { authRequired: true }, {
 			return RocketChat.API.v1.unauthorized();
 		}
 
+		// allow special handling of particular setting types
+		const setting = RocketChat.models.Settings.findOneNotHiddenById(this.urlParams._id);
+		if (setting.type === 'action' && this.bodyParams && this.bodyParams.execute) {
+			//execute the configured method
+			Meteor.defer(() => Meteor.call(setting.value));
+
+			return RocketChat.API.v1.success(); // we triggered the method - we won't wait for completion
+		}
+
+		if (setting.type === 'color' && this.bodyParams && this.bodyParams.editor && this.bodyParams.value) {
```

Comment:
This fixes #207 en-passant. @ThomasRoehl How about adding a downport of this code to RC?